Sources/SwiftFusion/Inference/VectorValues.swift

```diff
@@ -110,6 +99,53 @@ extension VectorValues: AdditiveArithmetic {
   public static var zero: VectorValues {
     return VectorValues()
   }
+
+  // MARK: - VectorProtocol conformance
+
+  public typealias VectorValuesSpaceScalar = Double
+
+  public mutating func add(_ x: Double) {
+    for index in _values.indices {
+      _values[index] += x
+    }
+  }
+
+  public func adding(_ x: Double) -> VectorValues {
+    var result = self
+    result.add(x)
+    return result
+  }
+
+  public mutating func subtract(_ x: Double) {
+    for index in _values.indices {
+      _values[index] -= x
+    }
+  }
+
+  public func subtracting(_ x: Double) -> VectorValues {
+    var result = self
+    result.subtract(x)
+    return result
+  }
+
+  public mutating func scale(by scalar: Double) {
+    for index in _values.indices {
+      _values[index] *= scalar
+    }
+  }
+
+  public func scaled(by scalar: Double) -> VectorValues {
+    var result = self
+    result.scale(by: scalar)
+    return result
+  }
+
+  // MARK: - Additional EuclideanVectorSpace requirements.
+
+  public var squaredNorm: Double {
```

Comment:
Do we still need the `.differentiableMap` methods?

Reply:
`.differentiableMap` is still necessary for writing a differentiable function that does a map over an array.

This function currently isn't declared `@differentiable`, so it does not need to use `.differentiableMap`.

I like to avoid adding the `@differentiable` annotation until we actually need to use a function in a differentiable way, to keep things as simple as possible.